apps/ledger-live-desktop/tests/specs/speculos/send.tx.spec.ts

```diff
@@ -1,48 +1,233 @@
+    },
+    async ({ app }) => {
+      await app.layout.goToAccounts();
+      await app.accounts.navigateToAccountByName(transactionsToken.accountToDebit.accountName);
+      await app.account.navigateToTokenInAccount(transactionsToken.token);
+      await app.account.clickSend();
+      await app.send.fillRecipient(transactionsToken.accountToCredit1.address);
+      await app.send.checkASAError();
+      await app.send.fillRecipient(transactionsToken.accountToCredit2.address);
+      await app.send.checkContinueButtonEnable();
+      await app.layout.checkInputErrorNotVisible();
+    },
+  );
+});
+
+test.describe("Check ETH input error", () => {
+  test.use({
+    userdata: "speculos-tests-app",
+    testName: `ETH_input_error_${transactionETH_BTC.accountToDebit.currency.uiName}`,
+    speculosCurrency:
+      specs[transactionETH_BTC.accountToDebit.currency.deviceLabel.replace(/ /g, "_")],
+    speculosOffset: 0,
+  });
+
+  test(
+    `Check [${transactionETH_BTC.accountToDebit.currency}] input error`,
+    {
+      annotation: {
+        type: "TMS",
+        description: "B2CQA-1904",
+      },
+    },
+    async ({ app }) => {
+      await app.layout.goToAccounts();
+      await app.accounts.navigateToAccountByName(transactionETH_BTC.accountToDebit.accountName);
+
+      await app.account.clickSend();
+      await app.send.fillRecipient(transactionETH_BTC.accountToCredit.address);
+      await app.send.checkInvalidAddressError(transactionETH_BTC);
+    },
+  );
+});
+
+for (const [i, transaction] of transactionsInputsInvalid.entries()) {
+  test.describe("Check error message for send test (invalid input)", () => {
     test.use({
       userdata: "speculos-tests-app",
-      testName: `sendApprove_${transaction.accountToDebit.currency.uiName}`,
-      speculosCurrency: specs[transaction.accountToDebit.currency.deviceLabel.replace(/ /g, "_")],
+      testName: `Send_test_invalid_input_${transaction.transaction.accountToDebit.currency.uiName}`,
+      speculosCurrency:
+        specs[transaction.transaction.accountToDebit.currency.deviceLabel.replace(/ /g, "_")],
       speculosOffset: i,
     });
 
     test(
-      `[${transaction.accountToDebit.accountName}] send Approve`,
+      `Check [${transaction.expectedErrorMessage}] for [${transaction.transaction.accountToDebit.currency.uiName}] - Send test invalid input - ${i} `,
       {
         annotation: {
           type: "TMS",
           description: "B2CQA-473",
         },
       },
       async ({ app }) => {
-        await addTmsLink(getDescription(test.info().annotations).split(", "));
+        await app.layout.goToAccounts();
+        await app.accounts.navigateToAccountByName(
+          transaction.transaction.accountToDebit.accountName,
+        );
+
+        await app.account.clickSend();
+        await app.send.fillRecipient(transaction.transaction.accountToCredit.address);
+        await app.send.clickContinue();
+        await app.send.fillAmount(transaction.transaction);
```

Comment:
you could pass tx amount directly here, it'd allow you to log amount in report in `fillAmount` step